### DISCUSSION TITLE
[SKIP]

### DISCUSSION THREAD

**Discussion by akauppi** [firebase/firebase-js-sdk#4637] [Archive] [answered]
I think the `@firebase` sub-packages use wrong hashes, when referring to `@firebase/app` as their `peerDependency`. Details for 0.900.17:

```
#@firebase/app/package.json:

  "name": "@firebase/app",
  "version": "0.0.900-exp.555fe23c5",    <-- the version we have
```

```
@firebase/auth/package.json:

  "name": "@firebase/auth",
  "version": "0.0.900-exp.0bd766e7a",
…
  "peerDependencies": {
    "@firebase/app": "0.0.900-exp.0bd766e7a" 
  },
```

Expected:
- the app version to be `0.0.900-exp.555fe23c5`

Actual:
- each sub-packages refers to `@firebase/app` with the same hash they themselves have.

I don't really know why things work, at all. Maybe the ending is not important for Node (it doesn't warn me of missing dependencies). If the ending doesn't matter, maybe everyone should just `peerDependencies: { "@firebase/app": "0.0.900-exp" }`, without the hash?)

**Comment by Feiyang1** [comment, marked as answer]
I think your `node_modules` might be out of sync. I got the correct version for `@firebase/app` after installing `0.900.17`:
```
  "name": "@firebase/app",
  "version": "0.0.900-exp.0bd766e7a",
```

Perhaps the lock file? Can you delete it and try a clean install?

**Reply by akauppi** [reply]
You're right. Human error.. Retreating.
